.github/workflows/slack-workflow-queue.yml

```diff
@@ -0,0 +1,59 @@
+name: Schedule Wait Queue Check
+
+# Check GitHub's API if there are any workflows in wait queue for longer then 5 minutes
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/slack-workflow-queue.yml'
+  schedule:
+    - cron: "*/5 * * * *"
+
+jobs:
+  check-queue:
+    name: Check Workflow Wait Queue
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Check for workflows in wait queue
+        id: check
+        shell: bash
+        run: |
+          set -euo pipefail
+
+          GITHUB_API_URL="https://api.github.com/repos/${{ github.repository }}/actions/runs"
+          WAIT_THRESHOLD_MINUTES=5
+          WAIT_THRESHOLD_SECONDS=$((WAIT_THRESHOLD_MINUTES * 60))
+
+          # Fetch workflow runs
+          WORKFLOW_RUNS=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" $GITHUB_API_URL)
+          # Check for queued workflows
+          CURRENT_TIME=$(date +%s)
+          WORKFLOWS_IN_QUEUE=$(echo $WORKFLOW_RUNS | jq -r '.workflow_runs[] | select(.status == "queued") | {id: .id, run_started_at: .run_started_at}')
+
+          set -x
+          MESSAGE=""
+          for WORKFLOW in $(echo "${WORKFLOWS_IN_QUEUE}" | jq -r '.id'); do
+            RUN_STARTED_AT=$(echo "${WORKFLOWS_IN_QUEUE}" | jq -r "select(.id == ${WORKFLOW}) | .run_started_at")
+            RUN_STARTED_TIME=$(date -d "$RUN_STARTED_AT" +%s)
+            TIME_IN_QUEUE=$((CURRENT_TIME - RUN_STARTED_TIME))
+
+            if [ $TIME_IN_QUEUE -ge $WAIT_THRESHOLD_SECONDS ]; then
+              WORKFLOW_URL="https://github.com/${{ github.repository }}/actions/runs/${WORKFLOW}"
+              MESSAGE="${MESSAGE} <${WORKFLOW_URL}|${WORKFLOW}>"
+            fi
+          done
+
+          if [ "$MESSAGE" !=  "" ]; then
+            MESSAGE=":alert: Workflows in Wait Queue [> ${WAIT_THRESHOLD_MINUTES} min]: ${MESSAGE}"
```

Comment:
I think it could be helpful to include this link in the message: https://github.com/dfinity/ic/actions?query=is%3Aqueued - so that people can directly see which jobs are queued.

Reply:
Indeed. I've made a hyperlink on "Workflows in Wait Queue".